### ISSUE TITLE
BTREE indexes to be case sensitive and GIN are case insensitive

### ISSUE THREAD

**Issue by wakibi** [devinit/ddw-analyst-ui#335]
By their nature, BTREE indexes are case sensitive and that should be included in our documentation. Removing the case sensitivity on BTREEs removes the performance gains they were used in the first place. Users need made be aware of this

**Comment by k8hughes** [comment]
I think we still need this.

**Comment by k8hughes** [comment]
To be able to work on this we'd need to know the collumns that are most searched by for it to be efficent.

**Comment by wakibi** [comment]
There is a blocker here, some of the fields to be indexed are rather too long and the DB can not index them. Suggestions on how to handle this are much welcome @akmiller01 @dean-breed @bill-anderson

**Comment by bill-anderson** [comment]
You don't want BTREE indexes on titles and descriptions, do you? In my old money (of my particular old currency) we had cross-reference indexes for free text fields. Is this GiST in new money?

**Comment by wakibi** [comment]
That's right @bill-anderson. GiST and GIN are to be used for those extremely long text columns, but they too have a max string length (which is about 8000 characters) they can be applied to. Unfortunately, some of our data exceeds that length!

**Comment by bill-anderson** [comment]
So what I think we should do is move this into an ETL discussion to limit all free-text fields to, say, 5000 characters. I don't think that will affect the integrity of the data. But we should do a bit of research on the long culprits.

**Comment by dean-breed** [comment]
@wakibi Would it be possible to get a list of reporting organisations and columns which would be truncated?

**Comment by dean-breed** [comment]
@wakibi Can I get this list?

**Comment by edwinmp** [comment]
@dean-breed sorry Chris is on leave ... let me stick this on Slack where he can't miss it.

**Comment by wakibi** [comment]
@dean-breed Find attached. I have added all those with more than 7500 (my reasoning is those approaching that figure could easily soon hit the 8000 ceiling as well).
[long_columns.csv](https://github.com/devinit/ddw-analyst-ui/files/8553982/long_columns.csv)

**Comment by dean-breed** [comment]
@wakibi - I could do something on this myself but to avoid mistake likelihood. Could you output the reporting org reference also?

**Comment by wakibi** [comment]
> @wakibi - I could do something on this myself but to avoid mistake likelihood. Could you output the reporting org reference also?

[long_columns.csv](https://github.com/devinit/ddw-analyst-ui/files/8554330/long_columns.csv)

Attached. I had forgotten to format the first one.

**Comment by dean-breed** [comment]
Hi Chris @wakibi, is it possible to output this list as part of the ETL process? I could then include the information in the data quality file?

**Comment by wakibi** [comment]
Hi @dean-breed, do you want it sent as an email attachment?

**Comment by dean-breed** [comment]
That would be great if it could. Or just stored in a repo somewhere. Either works @wakibi
